src/doc/trpl/documentation.md

```diff
@@ -1,224 +1,222 @@
+```
 
-rustdoc also supports personalizing the output from crates' documentation,
-similar to markdown options.
+This [unfortunate error](https://github.com/rust-lang/rust/issues/22547) is
+correct: documentation comments apply to the thing after them, and there's no
+thing after that last comment.
 
-- `--html-in-header FILE`: includes the contents of `FILE` at the
-  end of the `<head>...</head>` section.
-- `--html-before-content FILE`: includes the contents of `FILE`
-  directly after `<body>`, before the rendered content (including the
-  search bar).
-- `--html-after-content FILE`: includes the contents of `FILE`
-  after all the rendered content.
+[rc-new]: http://doc.rust-lang.org/nightly/std/rc/struct.Rc.html#method.new
 
-# Using the Documentation
+### Writing documentation comments
 
-The web pages generated by rustdoc present the same logical hierarchy that one
-writes a library with. Every kind of item (function, struct, etc) has its own
-color, and one can always click on a colored type to jump to its
-documentation. There is a search bar at the top, which is powered by some
-JavaScript and a statically-generated search index. No special web server is
-required for the search.
+Anyway, let's cover each part of this comment in detail:
 
-[hoedown]: https://github.com/hoedown/hoedown
+```
+/// Constructs a new `Rc<T>`.
+# fn foo() {}
+```
 
-# Testing the Documentation
+The first line of a documentation comment should be a short summary of its
+functionality. One sentence. Just the basics. High level.
 
-`rustdoc` has support for testing code examples which appear in the
-documentation. This is helpful for keeping code examples up to date with the
-source code.
+```
+///
+/// Other details about constructing `Rc<T>`s, maybe describing complicated
+/// semantics, maybe additional options, all kinds of stuff.
+///
+# fn foo() {}
+```
 
-To test documentation, the `--test` argument is passed to rustdoc:
+Our original example had just a summary line, but if we had more things to say,
+we could have added more explanation in a new paragraph.
 
-~~~ {.sh}
-rustdoc --test crate.rs
-~~~
+#### Special sections
 
-## Defining tests
+```
+/// # Examples
+# fn foo() {}
+```
 
-Rust documentation currently uses the markdown format, and rustdoc treats all
-code blocks as testable-by-default unless they carry a language tag of another
-language. In order to not run a test over a block of code, the `ignore` string
-can be added to the three-backtick form of markdown code block.
+Next, are special sections. These are indicated with a header, `#`. There
+are three kinds of headers that are commonly used. They aren't special syntax,
+just convention, for now.
 
-~~~md
 ```
-// This is a testable code block
+/// # Panics
+# fn foo() {}
 ```
 
-```rust{.example}
-// This is rust and also testable
-```
+If your function or method can panic, you should note the conditions in which
+it panics here. Any function in Rust can panic, so documentation is extra
```

Comment:
I don't quite follow why "any rust function can panic" means documentation is super important. (e.g. misapplying the logic, "any function in Rust can do a network request to download a 1GB file, so documentation is extra important.")

Maybe, something along the lines of:

> `panic!`s are often used for "contract violations", but are hard to recover from, so precisely documenting the preconditions a user needs to satisfy is extra important.

(I don't like my phrasing there, though.)